src/test/java/nl/thehyve/ocdu/ODMGenerationTests.java

```diff
@@ -0,0 +1,77 @@
+import nl.thehyve.ocdu.soap.ResponseHandlers.GetStudyMetadataResponseHandler;
+import nl.thehyve.ocdu.soap.ResponseHandlers.ListAllByStudyResponseHandler;
+import nl.thehyve.ocdu.validators.ClinicalDataOcChecks;
+import org.junit.Before;
+import org.junit.Test;
+import org.openclinica.ws.beans.StudySubjectWithEventsType;
+
+import javax.xml.soap.MessageFactory;
+import javax.xml.soap.MimeHeaders;
+import javax.xml.soap.SOAPMessage;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by jacob on 6/9/16.
+ */
+public class ODMGenerationTests {
+
+    MetaData metaData;
+    List<StudySubjectWithEventsType> testSubjectWithEventsTypeList;
+    OcUser testUser;
+    UploadSession testSubmission;
+    ClinicalDataFactory factory;
+
+    ClinicalDataOcChecks clinicalDataOcChecks;
+
+    Path testODMGenerationCorrect;
+
+    @Before
+    public void setUp() throws Exception {
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        File testFile = new File("docs/responseExamples/getStudyMetadata3.xml");
+        FileInputStream in = new FileInputStream(testFile);
+        SOAPMessage mockedResponseGetMetadata = messageFactory.createMessage(null, in);//soapMessage;
+        this.metaData = GetStudyMetadataResponseHandler.parseGetStudyMetadataResponse(mockedResponseGetMetadata);
+
+        this.testUser = new OcUser();
+        this.testUser.setUsername("tester");
+        this.testSubmission = new UploadSession("submission1", UploadSession.Step.MAPPING, new Date(), this.testUser);
+        this.factory = new ClinicalDataFactory(testUser, testSubmission);
+
+        this.testSubjectWithEventsTypeList = createStudySubjectWithEventList();
+
+        this.testODMGenerationCorrect = Paths.get("docs/exampleFiles/odmGeneration.txt");
+    }
+
+    private List<StudySubjectWithEventsType> createStudySubjectWithEventList() throws Exception {
+        File mockResponseListAllByStudyFile = new File("docs/responseExamples/listAllByStudyResponse.xml");
+        InputStream mockResponseListAllByStudyFileInputStream = new FileInputStream(mockResponseListAllByStudyFile);
+
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        SOAPMessage soapMessage = messageFactory.createMessage(new MimeHeaders(), mockResponseListAllByStudyFileInputStream);
+        return ListAllByStudyResponseHandler.retrieveStudySubjectsType(soapMessage);
+    }
+
+
+    @Test
+    public void testODMGeneration() throws Exception {
+        List<ClinicalData> correctClinicalData = factory.createClinicalData(testODMGenerationCorrect);
+        clinicalDataOcChecks = new ClinicalDataOcChecks(metaData, correctClinicalData, testSubjectWithEventsTypeList);
+        ODMService odmService = new ODMService();
+        String result = odmService.generateODM(correctClinicalData, metaData);
```

Comment:
It is probably pretty hard to check XML for correctness in the test. For now maybe we could start with a simple assertion checking if generated XML is well-formed (which would implicitly tests that it is not null and not empty as well). I found this nice SO question which could help us do it: http://stackoverflow.com/questions/6362926/xml-syntax-validation-in-java